sympy/series/formal.py

```diff
@@ -1210,6 +1401,59 @@ def __rmul__(self, other):
         return self.__mul__(other)
 
 
+class FiniteFormalPowerSeries(FormalPowerSeries):
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return Expr.__new__(cls, *args)
+
+    def __init__(self, *args):
+        pass
+
+    def _eval_term(self, pt):
+        try:
+            pt_xk = self.xk.coeff(pt)
+            pt_ak = S.Zero
+            if pt is not S.Zero:
+                pt_ak = self.ak[pt-1]  # Simplify the coefficients
+        except IndexError:
+            term = S.Zero
+        else:
+            term = (pt_ak * pt_xk)
+
+        if self.ind:
+            ind = S.Zero
+            for t in Add.make_args(self.ind):
+                pow_x = self._get_pow_x(t)
+                if pt == 0 and pow_x < 1:
+                    ind += t
+                elif pow_x >= pt and pow_x < pt + 1:
+                    ind += t
+            term += ind
+
+        return term.collect(self.x)
+
+    def polynomial(self, n=6):
+        """Truncated series as polynomial.
+
+        Returns series expansion of ``f`` upto order ``O(x**n)``
+        as a polynomial(without ``O`` term).
+        """
+        if (n > len(self.ak)+1):
+            raise ValueError("The order of truncate should be less than or equal to the"
+                "order specified in the compose function.")
+        terms = []
+        for i, t in enumerate(self):
+            xp = self._get_pow_x(t)
+            if xp >= n:
+                break
+            elif xp.is_integer is True and i == n + 1:
+                break
+            elif t is not S.Zero:
+                terms.append(t)
+
+        return Add(*terms)
```

Comment:
Can we use, `Add.fromiter(terms)`? May be I asked for this earlier too and then someone, said that it is not possible to do so. If that's the case, then feel free to resolve this, otherwise do the replacement.